### ISSUE TITLE
WebApp:az webapp plan show` returns exit code 0 for non-existing resource

### ISSUE THREAD

**Issue by MarkAtAgilliance** [Azure/azure-cli#16436]
## Describe the bug
az webapp plan show` returns exit code 0 for non-existing resource.  Other commands return a non-zero return code.

This behavior of "show" needs to be consistent across the CLI or script writing/maintenance is going to become unmanageable having to remember which commands bahave one way vs another.

Possibly a similar issue to #14572

**Command Name**
`az appservice plan show`

**Errors:**
Returns 0 if app service plan does not exist.

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az appservice plan show -n {"non-existant-plan-name"} -g {"existing-RG"}`

## Expected Behavior
returns non-zero result

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.17.1

Extensions:
application-insights 0.1.13

```

**Comment by yonzhan** [comment]
webapp

**Comment by ThejaChoudary** [comment]
Able to repro the issue.Will look into this

**Comment by cocowalla** [comment]
Is there any workaround in the meantime?

**Comment by MarkAtAgilliance** [comment]
My workaround was to check the contents of the returned json.   It is $null when the plan does not exist where specified (including incorrect RG) however this forces me to check I have the correctly named RG first.

BTW, my original title was wrong - the command uses "appservice plan" not "webapp plan".  i.e. az appservice plan show -n ... -g ...

**Comment by cocowalla** [comment]
I ended up going for:

```bash
plan_name="my-plan"

if [[ $(az appservice plan list --query "[?name=='${plan_name}'].name" -o tsv | wc -l) -gt 0 ]]; then
  # App Service plan already exists
fi
```

It is quite annoying coming across these inconsistencies in the Azure CLI from time to time. I get it's vast, with a *lot* of commands, but the structure is consistent, and so it feels like consistency of behaviour should be achievable - @ThejaChoudary have you (as in the Az CLI team 😄) looked into the possibility of creating some kind of linter/analyser? Idea is you feed it rules (like "any show command should return non-zero if resource not found").

**Comment by Kotasudhakarreddy** [comment]
Duplicate of #17424

**Comment by MarkAtAgilliance** [comment]
I'd argue 17424 is a duplicate of this.

**Comment by Kotasudhakarreddy** [comment]
Yeah I will agree with you @MarkAtAgilliance. I have given fix for #17424 and later on I have figure out there is an issue already logged for it. So thought of making this issue as duplicate as I cannot change the Fix history for #17424. Sorry for marked this as a duplicate. Hope you convince :)
